agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java

```diff
@@ -528,6 +528,15 @@ private void exportRequest(SpanData span) {
         requestData.setId(span.getSpanId());
         requestData.getContext().getOperation().setId(span.getTraceId());
 
+        String locationIp = attributes.get(SemanticAttributes.HTTP_CLIENT_IP);
+        if (locationIp == null) {
```

Comment:
empty check?

Reply:
in this context, null means absent, and "" means (for some strange reason) user intentionally set "", so makes sense(-ish) to pass it on